src/Azure.php

```diff
@@ -62,11 +64,65 @@ public function handle($request, Closure $next)
         return $this->handlecallback($request, $next, $access_token, $refresh_token);
     }
 
+    /**
+     * Handle an incoming request in a testing environment
+     * Assumes tester is calling actingAs or loginAs during testing to run this correctly
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param Closure $next
+     * @return \Illuminate\Http\RedirectResponse|\Illuminate\Routing\Redirector|mixed
+     */
+    protected function handleTesting(Request $request, Closure $next)
+    {
+        $user = Auth::user();
+
+        if (!isset($user))
+        {
+            return $this->redirect($request, $next);
+        }
+
+        return $this->handlecallback($request, $next);
+    }
+
+    /**
+     * Gets the azure url
+     *
+     * @return String
+     */
+    public function getAzureUrl()
+    {
+        return $this->baseUrl . env('AZURE_TENANT_ID') . $this->route . "authorize?response_type=code&client_id=" . env('AZURE_CLIENT_ID') . "&resource=" . urlencode(env('AZURE_RESOURCE'));
+    }
+
+    /**
+     * Redirects to the Azure route
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\RedirectResponse|\Illuminate\Routing\Redirector|mixed
+     */
     public function azure(Request $request)
     {
-        return redirect()->away( $this->baseUrl . env('AZURE_TENANT_ID') . $this->route . "authorize?response_type=code&client_id=" . env('AZURE_CLIENT_ID') . "&resource=" . urlencode(env('AZURE_RESOURCE')) );
+        return redirect()->away( $this->getAzureUrl() );
+    }
+
+    /**
+     * Customized Redirect method
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\RedirectResponse|\Illuminate\Routing\Redirector|mixed
+     */
+    protected function redirect(Request $request)
```

Comment:
Will this always redirect to the login route? If so, we could name it such so it's not confusing between `redirect()` and `$this->redirect()`.

Reply:
@jaredrhodes In the base it does.  However, it can be overwritten to redirect wherever.

Here's an example in a production app:

```
    protected function redirect(Request $request)
    {
        $user = Auth::user();

        if ($user !== null && $user->isRootEmployee())
        {
            return $this->azure($request);
        }
        else
        {
            return parent::redirect($request);
        }
    }
```

If the naming is still confusing, what do you suggest?

Reply:
I guess since the point is for it to allow for branching `redirect()` is fine.